pkg/datapath/linux/node.go

```diff
@@ -1010,7 +1011,7 @@ func (n *linuxNodeHandler) nodeUpdate(oldNode, newNode *nodeTypes.Node, firstAdd
 		return errs
 	}
 
-	if n.nodeConfig.EnableAutoDirectRouting {
+	if n.nodeConfig.EnableAutoDirectRouting && !n.enableEncapsulation(newNode) {
```

Comment:
I think we don't need this because in the datapath, tunnel redirection happens before kernel routing.